Runtime/RealtimeDatabase/RealtimeDatabase.cs

```diff
@@ -886,7 +886,7 @@ object GenerateBody(object value)
                 string paramString = value.ToString();
                 int length = paramString.Length;
                 //make sure it's not json, if json, leave as it is.
-                if (paramString[0] != '{' && paramString[length - 1] != '}')
+                if (paramString[0] != '{' && paramString[length - 1] != '}' && paramString[0] != string.Empty)
```

Comment:
`paramString[0]` will be evaluated to a char. And, `string.Empty` is a string type, We cannot compare a char with a string directly. 

However, there is a caveat that was not mentioned in the issue. For a single char / empty string, we still need to wrap it with quotes. So, it would be better to form the condition like this - `paramString.Length <= 1 || (paramString[0] != '{' && paramString[length - 1] != '}')`. In the case of a string with less than or equal to 1 char, it will do a [short-circuit evaluation](https://www.geeksforgeeks.org/short-circuit-evaluation-in-programming/) so there is no chance of getting `IndexOutOfRangeException` and it will generate our expected output.